internal/cmd/internal/olmv1/operator_update.go

```diff
@@ -0,0 +1,40 @@
+package olmv1
+
+import (
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	"github.com/operator-framework/kubectl-operator/internal/cmd/internal/log"
+	v1action "github.com/operator-framework/kubectl-operator/internal/pkg/v1/action"
+	"github.com/operator-framework/kubectl-operator/pkg/action"
+)
+
+// NewOperatorUpdateCmd allows updating a selected operator
+func NewOperatorUpdateCmd(cfg *action.Configuration) *cobra.Command {
+	i := v1action.NewOperatorUpdate(cfg)
+	i.Logf = log.Printf
+
+	cmd := &cobra.Command{
+		Use:   "operator <operator>",
+		Short: "Update an operator",
+		Args:  cobra.ExactArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			i.Package = args[0]
+			_, err := i.Run(cmd.Context())
+			if err != nil {
+				log.Fatalf("failed to update operator: %v", err)
+			}
+			log.Printf("operator %q updated", i.Package)
+		},
+	}
+	bindOperatorUpdateFlags(cmd.Flags(), i)
+
+	return cmd
+}
+
+func bindOperatorUpdateFlags(fs *pflag.FlagSet, i *v1action.OperatorUpdate) {
+	fs.StringVar(&i.Version, "version", "", "desired operator version (single or range) in semver format. AND operation with channels")
+	fs.StringArrayVar(&i.Channels, "channels", []string{}, "desired channels for operator versions. AND operation with version")
+	fs.StringVar(&i.UpgradeConstraintPolicy, "upgrade-constraint-policy", "", "controls whether the upgrade path(s) defined in the catalog are enforced, one of CatalogProvided|SelfCertified), Default: CatalogProvided")
+	fs.BoolVar(&i.OverrideUnset, "override-unset-with-current", false, "when enabled, any unset flag value will be overridden with value already set in current operator")
+}
```

Comment:
do we need an option to provide a new serviceaccount name?

Reply:
good idea, I'll add it the same way `catalog` creation has it now

Reply:
well, it wasn't sorted out when you commented because I didin't yet make the code change.

Reply:
+1 for @akihikokuroda thought.
I do not think we should allow a user to pass the SA for upgrade an ClusterExtension
What is the use case?
Why would it be required?

Reply:
After thinking more, I think it's OK, do not add the catalog either WHY?
- We will upgrade what is installed, correct?
- So should be from the source used previously, not the other. 
- Either we cannot have 2 instances of the same solution so I think we can ignore this idea of catalog 👍

Reply:
I think the only way we have of specifying a catalog to install from within the clusterExtension CRD is through an optional labelSelector. That being said, I think it's best to support modifying at least that for now, even if we don't support specifying individual catalogs yet.

Reply:
I added full selector support, courtesy of finding the `metav1.ParseToLabelSelector`

Unit test case was not extended to show it, because we'll be switching to using `WithInterceptorFuncs` anyway in https://github.com/operator-framework/kubectl-operator/pull/226/files

I added additional manual testing output in the walkthrough though